allow mirror parent node to be configured

benib opened the pull request:
This is helpful if styles/selectors need to be scoped to a container. If mirror element gets appended to `body` then all styles and selectors need to match every element within body.

bevacqua commented:
Can't you just use a CSS class?

benib commented:
The thing I write gets embedded within a website I have no control of. I don't want to add css with selectors that can possibly mess around with the surrounding website. So I wanted to scope everything within one container. Using an iframe is not a nice option either.

bevacqua commented:
I can see the value in merging this. Couple of points:

- I would rename this as `mirrorContainer`, it's more concise
- Can you verify that appending the mirror somewhere other than the `body` doesn't affect the CSS? (Besides your custom thing, if you were to just palce it somewhere else for the fun of it)

benib commented:
I changed the option name to `mirrorContainer`, makes sense to me.

As for affecting the CSS, I'm not sure if I got your question. The CSS that is in the dragula project works perfectly as long as you don't mess it up with other selectors matching the mirror element or any other element that gets generated by dragula. But this is the case even if the element is appended directly to `body`.
As the mirror has `position: fixed` its position is relative to the viewport anyway, no matter where it is placed.

If I completely missed your point, please clearify.

bevacqua commented:
Cool. Do me a favor and squash the commits, and I'll merge it. Thanks

bevacqua commented on readme.markdown:
```diff
@@ -68,11 +68,12 @@ dragula(containers, {
+  revertOnSpill: false,              // spilling will put the element back where it was dragged from, if this is true
+  removeOnSpill: false,              // spilling will `.remove` the element, if this is true
+  delay: false                       // enable regular clicks by setting to true or a number of milliseconds
+  mirrorContainer: document.body // set the element that gets mirror elements appended
```
also, please fix indentation here

bevacqua commented on readme.markdown:
```diff
@@ -158,6 +159,10 @@ Number of milliseconds during which clicks where the mouse button is released wi
 
+#### `options.mirrorContainer`
+
+You can provide an `Node` that will be the parent of all generated mirror nodes.
```
Just state DOM element instead of `Node`. Let's use this copy:

> The DOM element where the mirror element displayed while dragging will be appended to. Defaults to `document.body`.

benib commented:
Sorry for the delay. Copy changed, indent fixed, commits squashed. If there is anything else, let me know.

benib commented:
@bevacqua I don't want to pressure you, but if there is anything I could/should do to get this merged, let me know.

bevacqua commented:
Could you add some tests and then squash the commits? Thanks!

benib commented:
alright, addded tests and squashed. can't rewrite history across the merge though.
